### PR TITLE
Migrate to comrak/Rust

### PR DESCRIPTION
**Pull request by gjtorikian**
Will close https://github.com/gjtorikian/commonmarker/issues/107

pure fyi to @kivikakk for demo of how to integrate Rust into Ruby. I need to study more on how to package the gem _itself_ as a precompiled artifact for various OSes ([which Nokogiri recently did](https://github.com/sparklemotion/nokogiri/releases/tag/v1.11.0)). 

For anyone else reading this: node iteration/manipulation will probably not be supported during the major release cut over off of C and onto Rust. I'd like to incorporate https://github.com/gjtorikian/selma into https://github.com/gjtorikian/html-pipeline and encourage using a post-processor. Part of this stems from the fact that implementing an AST parser in Ruby is slow; selma also uses a Rust create to more efficiently change multiple nodes at once (via CSS selectors). 

So, basically, I'm working on all three of these projects at once, but they won't be released until I can manage to make precompiled gems happen.

### CODE REVIEW

**Comment by kivikakk** [comment]
Looking awesome so far. `rake test` working well locally.

Is it ridiculously presumptuous of me to add commits to this branch? Drop them if you have any reservations at all, just acting on my compulsions a little bit; removed the out of date Makefile in a76379a, and added some casts for `rb_hash_foreach` calls to build without warnings in aa4afde (on arm64-darwin anyway).

**Comment by gjtorikian** [comment]
> Is it ridiculously presumptuous of me to add commits to this branch? 

Not at all! I usually have only a few hours a week free to work on this so any help is appreciated.

File: ext/commonmarker/extconf.rb [hunk 1]
```diff
@@ -51,17 +47,9 @@ def darwin?
...
-    tarball_extract_path = if darwin?
-      recipe.host =~ /arm64-apple-darwin(\d+)\.\d+\.0/
-      host = "arm64-darwin#{$1}"
-      File.join(GEM_ROOT_DIR, "tmp", host, "commonmarker", RUBY_VERSION, "tmp", recipe.host, "ports", recipe.name, recipe.version, "#{recipe.name}-#{recipe.version}")
-    else
-      File.join(GEM_ROOT_DIR, "tmp", recipe.host, "ports", recipe.name, recipe.version, "#{recipe.name}-#{recipe.version}")
-    end
-
-    # Why is this so long?
+    tarball_extract_path = File.join("tmp", recipe.host, "ports", recipe.name, recipe.version, "#{recipe.name}-#{recipe.version}")
```
Comment (gjtorikian) [inline ext/commonmarker/extconf.rb:50]: @kivikakk I'm real emberassed you had to see all this, hah

**Comment by gjtorikian** [comment]
PR looks great. Truly, thanks so much for your help. 

In a subsequent (final?) PR, I'm going to use https://github.com/rake-compiler/rake-compiler-dock to precompile native gems.
